recipes/pybind11/all/conanfile.py

```diff
@@ -40,16 +50,24 @@ def package(self):
         self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
         cmake = self._configure_cmake()
         cmake.install()
-        os.unlink(os.path.join(self.package_folder, "lib", "cmake", "pybind11", "pybind11Config.cmake"))
-        os.unlink(os.path.join(self.package_folder, "lib", "cmake", "pybind11", "pybind11ConfigVersion.cmake"))
+        lib_folder = os.path.join(
+            self.package_folder, "lib", "cmake", "pybind11")
+        os.rename(
+            os.path.join(lib_folder, "pybind11Config.cmake"),
+            os.path.join(lib_folder, "pybind11Install.cmake"))
+        os.unlink(os.path.join(lib_folder, "pybind11ConfigVersion.cmake"))
 
     def package_id(self):
         self.info.header_only()
 
     def package_info(self):
-        self.cpp_info.includedirs.append(os.path.join(self.package_folder, "include", "pybind11"))
+        self.cpp_info.includedirs.append(os.path.join(
+            self.package_folder, "include", "pybind11"))
 
         cmake_base_path = os.path.join("lib", "cmake", "pybind11")
         self.cpp_info.builddirs = [cmake_base_path]
-        self.cpp_info.build_modules = [os.path.join(cmake_base_path, "FindPythonLibsNew.cmake"),
-                                       os.path.join(cmake_base_path, "pybind11Tools.cmake")]
+
+        def get_path(filename):
+            return os.path.join(cmake_base_path, filename)
+        self.cpp_info.build_modules = [get_path("FindPythonLibsNew.cmake"),
+                                       get_path("pybind11Install.cmake")]
```

Comment:
Can't it be modeled with conan components? This block defeats CCI policy.

Reply:
I'm so happy that there is movement on this topic, I was just about to open a PR to bump the version to 2.6 since 2.5 packaged with conan does not expose all the custom targets and it looks like 2.6 changed this.
According to the docs, pybind11 exposes a lot of targets. https://pybind11.readthedocs.io/en/stable/compiling.html#advanced-interface-library-targets

Should all of these be modeled in the recipe?

Reply:
I would be better. Those 2 custom cmake functions `pybind11_strip()` and `pybind11_extension()` are in `pybind11Config.cmake`?